### PR TITLE
Bulk download to db

### PR DESCRIPTION
**Pull request by andrewjtdunn**
Takes manually downloaded bulk CSV file of comments and uploads them to the comments table in the database

### CODE REVIEW

**Comment by andrewjtdunn** [comment]
@michplunkett these tests pass locally but not when on Github. I'm missing something about how to to best use mocker/patch/assert_called_once_with in the complexity of functions within functions. Any insight you have is appreciated!

File: civiclens/tests/test_upload_bulk_csvs_to_db.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+def test_extract_fields_from_row_no_data():
+    try:
+        upload_bulk_csvs_to_db.extract_fields_from_row("", "test_id")
+    except Exception as e:
+        assert str(e) == "string indices must be integers, not 'str'"
+
+
+def test_extract_fields_from_row_good_data():
+    pl_row = {
```
Comment (michplunkett) [inline civiclens/tests/test_upload_bulk_csvs_to_db.py:93]: From a readability perspective, it'd be better to save this in a file as opposed to directly putting it in the python 🐍 file. Add at data folder to your `tests` folder and store it there.

**Comment by michplunkett** [comment]
Based on the error, it looks like you're definitely still calling the API. When you're on your local the API key will be there, so there won't be any error and it will call the service (though you may not see it). Based on just a cursory look, I'd check to see where you're establishing that connection with `https://api.regulations.gov` and what all other places are calling that API outside of the exclusive function you're testing. Swing through on Friday morning and we can debug this together if you haven't sorted it out by then.

```
'error': {'code': 'API_KEY_MISSING', 'message': 'No api_key was supplied. Get one at [https://api.regulations.gov:443](https://api.regulations.gov/)'}}
```

File: civiclens/regulations/models.py [hunk 2]
```diff
@@ -73,7 +73,4 @@ class Comment(models.Model):
     restrict_reason_type = models.CharField(max_length=100, blank=True, null=True)
     submitter_rep = models.CharField(max_length=100, blank=True, null=True)
     submitter_rep_address = models.CharField(max_length=255, blank=True, null=True)
-    submitter_rep_city_state = models.CharField(max_length=100, blank=True, null=True)
-
-class Search(models.Model):
-    searchterm = models.CharField(max_length=100)
+    submitter_rep_city_state = models.CharField(max_length=100, blank=True, null=True)
```
Comment (michplunkett) [inline civiclens/regulations/models.py:76]: It looks like your end of line formatter isn't working for one reason or another.

Comment (andrewjtdunn) [reply civiclens/regulations/models.py:76]: @abejburton fyi

**Review by rezarzky — CHANGES_REQUESTED** [review]
conflict on move_data_from_api_to_database and test_access_api_data

**Comment by andrewjtdunn** [comment]
> conflict on move_data_from_api_to_database and test_access_api_data

@rezarzky what's the issue? I just rebased from main and have no issues

**Review by rezarzky — APPROVED** [review]
assuming for move_data_from_api_to_database, document_data["id"] instead of data_for_db

File: civiclens/collect/upload_bulk_csvs_to_db.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        # Convert the parsed datetime object to the desired format string
+        formatted_datetime_str = parsed_datetime_dt.strftime("%Y-%m-%dT%H:%M:%SZ")
+        return formatted_datetime_str
+    except Exception as e:
+        print(f"Error parsing datetime string '{datetime_str}': {e}")
+        raise e
+
+
+# Test the function
+formatted_date = format_date("2024-03-06T05:00Z")
```
Comment (GregoryHo88) [inline civiclens/collect/upload_bulk_csvs_to_db.py:64]: We've got a few functions to format datetime. Might be good to collect the functions and put them under one class.

**Review by GregoryHo88 — APPROVED** [review]
I see no issues. The comment I had is for consideration - A backlog issue

File: civiclens/tests/test_access_api_data.py [hunk 4]
```diff
@@ -11,13 +10,13 @@ def test_is_duplicated_on_server():
...
 
 
-def test_is_duplicated_on_server_real_data():
-    # NRCS-2009-0004-0003 is known duplicated comment ID
-    assert access_api_data.pull_reg_gov_data(
-        api_key,
-        "comments",
-        params={"filter[searchTerm]": "NRCS-2009-0004-0003"},
-    )
+# def test_is_duplicated_on_server_real_data():
```
Comment (michplunkett) [inline civiclens/tests/test_access_api_data.py:13]: Generally, when merging code into `main` you should remove commented code.

File: civiclens/collect/upload_bulk_csvs_to_db.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    """
+    Loads csv file into a polars df of comments and filters out null comments
+    """
+    q = pl.scan_csv(file_name).filter(~pl.col("Comment").is_null())
+    df = q.collect()
+
+    return df
+
+
+def get_document_objectId(doc_id: str) -> str:
```
Comment (michplunkett) [inline civiclens/collect/upload_bulk_csvs_to_db.py:28]: 🔬 `get_document_object_id` is more pythonic. Right now it's a mix of snake and camel case.

Comment (andrewjtdunn) [reply civiclens/collect/upload_bulk_csvs_to_db.py:28]: in this case, objectId is a specific field so i opted to stay closer to the name in the data

File: civiclens/collect/upload_bulk_csvs_to_db.py (hunk 3 above)
Comment (andrewjtdunn) [reply civiclens/collect/upload_bulk_csvs_to_db.py:64]: @GregoryHo88 I made that into a backlogged issue

File: civiclens/collect/upload_bulk_csvs_to_db.py (hunk 5 above)
Comment (michplunkett) [reply civiclens/collect/upload_bulk_csvs_to_db.py:28]: Makes sense!

**Pull request merged**